Model/Api/UpdateCartCommon.php

```diff
@@ -398,6 +410,25 @@ public function createPayloadForVirtualQuote($quote, $requestData)
         ];
         return json_encode((object)$payload);
     }
+    
+    /**
+     * Create shipping assignment for shopping cart
+     *
+     * @param CartInterface $quote
+     */
+    public function setShippingAssignments($quote)
```

Comment:
This function sets cart extensions as well. Is it better to rename it?

Reply:
This is actually a private function copied from M2 core https://github.com/magento/magento2/blob/3c5ff49c11eec2d4510b25571acf0d30b804cdf1/app/code/Magento/Persistent/Model/QuoteManager.php#L225, so I follow its name.

Reply:
Ok, makes sense